### ISSUE TITLE
Application Crash at Contact Search

### ISSUE THREAD

**Issue by robberer** [signalapp/Signal-Android#5308]
I have:
- [x] searched open and closed issues for duplicates
- [x] read https://github.com/WhisperSystems/Signal-Android/wiki/Submitting-useful-bug-reports
### Bug description

When starting a new conversation the App crashes while type a letter in the search box.
### Steps to reproduce
- set signal as default sms app so it uses the phone contacts as well (important)
- start signal
- start new conversation
- type a letter in the search box
- app is crashing
### Device info

 **Model:** Meizu MX4

 **Android version:** 4.4.2

 **Signal version:** 3.12.0
### Link to debug log

adb logcat of the hopefully relevant section.

https://gist.github.com/robberer/12d9e645f90c0b459985

**Comment by KarolTrzeszczkowski** [comment]
I reproduced it on 
Meizu m2 note,
Flyme OS 4.5.4.2I, 
Android:5.1
Signal 3.14.2

It moves me to a password screen, when i type a letter in contacts search for new message.

**Comment by KarolTrzeszczkowski** [comment]
And now It crashes also when i try to switch  it back from being default SMS app.

**Comment by cascheberg** [comment]
@KarolTrzeszczkowski please provide a debug log, and file a new issue for bugs other than the original crash during search.

From @robberer's log:
```
java.lang.NullPointerException
        at android.database.AbstractCursor.getColumnIndex(AbstractCursor.java:287)
        at android.database.AbstractCursor.getColumnIndexOrThrow(AbstractCursor.java:301)
        at org.thoughtcrime.securesms.contacts.ContactSelectionListAdapter.onBindItemViewHolder(ContactSelectionListAdapter.java:117)
        at org.thoughtcrime.securesms.contacts.ContactSelectionListAdapter.onBindItemViewHolder(ContactSelectionListAdapter.java:52)
        at org.thoughtcrime.securesms.database.CursorRecyclerViewAdapter.onBindViewHolder(CursorRecyclerViewAdapter.java:148)
...
```

Vendor-specific Android bug? I feel like there was a very similar one here before, but I could not find it.

**Comment by KarolTrzeszczkowski** [comment]
My debug log with this bug and one from #5373

https://gist.github.com/dff65ac6cc0972b854c2

I actually did it twice - I reinstalled Signal.

**Comment by cascheberg** [comment]
```
java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.String.equalsIgnoreCase(java.lang.String)' on a null object reference
 	at android.database.AbstractCursor.getColumnIndex(AbstractCursor.java:287)
 	at android.database.AbstractCursor.getColumnIndexOrThrow(AbstractCursor.java:301)
 	at org.thoughtcrime.securesms.contacts.ContactSelectionListAdapter.onBindItemViewHolder(ContactSelectionListAdapter.java:117)
 	at org.thoughtcrime.securesms.contacts.ContactSelectionListAdapter.onBindItemViewHolder(ContactSelectionListAdapter.java:52)
    at org.thoughtcrime.securesms.database.CursorRecyclerViewAdapter.onBindViewHolder(CursorRecyclerViewAdapter.java:148)
...
```

**Comment by robberer** [comment]
I want to inform you that I can't provide further information to this bug as I'm not using the Meizu nor Signal anymore because I don't want to use Google Services, especially com.google.android.gms.

**Comment by hellenicsun** [comment]
I'm having the same issue, with my device:

Meizu M2 Note
Flyme OS 5.1.8.0G
Android 5.1
Latest Signal version

https://gist.github.com/595ba059194e7108a96ed5731360086c

**Comment by tgrrr** [comment]
Same issue here:

Meizu MX5.1
Flyme IS 5.1.2.0U
Not a custom ROM
Downloaded from app store
Signal 3.17.0

[](https://gist.github.com/anonymous/0112b0f85d428cbf06232daeb3460356)

**Comment by dkieffer** [comment]
Consistently crashes after typing a single letter in any contact search.

OnePlus 3
OxygenOS 3.2.4
Android 6.0.1
Signal 3.18.3 (and with 3.17.0)

**Comment by tgrrr** [comment]
Update: 

I've tested on 3 Meizu MX5 (with different google play logins, keyboards, etc. And it's limited to Signal on these devices.

Is there an update on fixing this issue?

**Comment by tgrrr** [comment]
Issue still exists for Flyme 6

SMS works, searching address book crashes Signal after a few seconds

https://gist.github.com/e68570d784adc7c7a0b7cc8486b68eb1

Let me know if more debugging data is necessary.

Cheers,

Phil

**Comment by McLoo** [comment]
@tgrrr there is no crash in the log, try capturing immediately after the crash occurs please.

**Comment by williamdes** [comment]
Same issue with my Meizu M5,

Model: MeizuM5
FlymeOS: 5.2.10
Android version: 6.0
Signal version: 4.7.4

https://gist.github.com/anonymous/8da0e975a7b7e44d04040aaa7a7e2e76

**Comment by dkieffer** [comment]
I didn't notice when, but I believe this issue has been fixed.

**Comment by williamdes** [comment]
Not on my Meizu :(
Signal version: 4.10.12

**Comment by williamdes** [comment]
Fixed with FlymeOS 6.7.8.8G BETA

**Comment by automated-signal** [comment]
GitHub Issue Cleanup:
See #7598 for more information.
